client/src/pages/login/index.js

```diff
@@ -1,52 +1,135 @@
-import React, { Component } from 'react';
-import { Redirect } from 'react-router-dom';
+import React, {
+  Component
+} from 'react';
+import {
+  Redirect
+} from 'react-router-dom';
+import SweetAlert from 'react-bootstrap-sweetalert';
 
-import { Input, Button } from '../../components';
+import {
+  Input,
+  Button
+} from '../../components';
 import './login.css';
 
 class Login extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      textInputValue: '',
-      isLogin: false
+      textInputValueUsername: '',
+      textInputValuePassword: '',
+      isLogin: false,
+      alert: null
     };
 
-    this.handleTextInputChange = this.handleTextInputChange.bind(this);
+    this.handleUsernameInputChange = this.handleUsernameInputChange.bind(this);
+    this.handlePasswordInputChange = this.handlePasswordInputChange.bind(this);
     this.handleLogin = this.handleLogin.bind(this);
+  }
+
+  errorPassOrUsername() {
+    const getAlert = () => (<SweetAlert
+      warning title = 'Wrong!'
+      confirmBtnBsStyle = 'danger'
+      cancelBtnBsStyle = 'danger'
+      onConfirm = {
+        () => this.hideAlert()
+      } >
+      Invalied Password or Username!
+    </SweetAlert>
+    );
+
+    this.setState({
+      alert: getAlert()
+    });
+  }
 
+  hideAlert() {
+    console.log('Hiding alert...');
+    this.setState({
+      alert: null
+    });
   }
 
   handleLogin(e) {
     e.preventDefault();
-    this.loginStatus();
+    const {
+      textInputValueUsername,
+      textInputValuePassword
+    } = this.state;
+    const data = JSON.stringify({
+      textInputValueUsername,
+      textInputValuePassword
+    });
+    fetch('/login_post', {
```

Comment:
Here's a new set of keywords I learned recently, which you could use here (or in the future with promises):
`async` and `await` - if you declare your function with "async", you can use "await" inside it, to wait for promises without needing to write "then" functions.  It's a new ES2017 thing, but you're using babel-preset-env so you are already set up in your build steps to write this style.

For example:
```
async handleLogin(e) {
  ...
  const response = await fetch('/login_post', { .... })
  //there's no then() needed!
  responseBody = response.json()
  ...
}
```